R/utils.R

```diff
@@ -403,3 +403,80 @@ defunction <- function(x) {
 get_unique_labels <- function(labels) {
   make.unique(gsub("[^[:alnum:]]", "_", tolower(labels)), sep = "_")
 }
+
+#' Create `renv` `.lock` file
+#'
+#' Function allows to create `renv::snapshot()` during the `teal` initialization.
+#'
+#' The `.lockfile` creation is based on [renv::snapshot()]. It allows multiple ways of `.lockfile` creation.
+#'
+#' - **Pre-computed `.lockfile`**: User is able to provide his own pre-computed `.lockfile` by setting the path to the
+#' `.lockfile` through `options(teal.renv.lockfile = "")`. Then this function is not used.
+#' - **Working directory `.lockfile`**: If `options(teal.renv.lockfile)` is empty, by default `teal` will
+#' create an `implicit` type of the `.lockfile`, that uses `renv::dependencies()` to detect all R packages in the
+#' current project working directory.
+#' - **`DESCRIPTION` based `.lockfile`**: You can always include a `DESCRIPTION` file in your working directory and
+#' enable `.lockfile` creation based on this file. To do this, run `renv::settings$snapshot.type("explicit")`.
+#' Naming of `type` is the same as in `renv::snapshot()`. For the `"explicit"` type refer to
+#' `renv::settings$package.dependency.fields()`
+#' to see what `DESCRIPTION` fields are included in the `.lockfile`.
+#' - **Custom files based `.lockfile`**: If you want to specify custom files as a base for the `.lockfile`, then run
+#' `renv::settings$snapshot.type("custom")` and set `renv.snapshot.filter` option.
+#'
+#' @note
+#' This function computes the `.lockfile` as a `promises::future_promise` promise, while
+#' running the evaluation of the process on a separate worker. If `future::plan()` was set to something different than
+#' `future::sequential`, it reuses the parallel backed. If not, it sets `future::multisession` as a parallel plan with
+#' 2 workers. Then, `shiny::ExtendedTask()` is used to run asynchronous computations. If `teal` app started with
+#' `future::sequential` plan, is it set back once the task is finished.
+#'
+#' @section `.lockfile` usage:
+#' Once you have a `.lockfile` file, you can restore R project with `ren::init();renv::restore()`.
+#'
+#' @param close (`logical(1)`) Flag that indicates whether to set `future::plan` to `future::sequential`
+#' when finishing the task.
+#'
+#' @seealso [renv::snapshot()], [renv::restore()].
+#'
+#' @return (`character(1)`) the path to the `lockfile` created in a `tempdir()`.
```

Comment:
this conflicts with the code comment below:
```
    # Below is not a file in tempdir() directory.
    # If a file is created in tempdir() it gets deleted on 'then(onFulfilled' part.
```

Reply:
Yeah, already changed. I was also reviewing it again. Now it states

```r
#' @return Nothing is returned. The side effect of this function is the creation of
#' `getOption("teal.internal.renv.lockfile")` file, which will be shared through the `teal` app.
```